src/app/zap-templates/zcl/tv-channel-cluster.xml

```diff
@@ -0,0 +1,80 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<configurator>
+  <domain name="CHIP"/>
+  <cluster>
+    <domain>General</domain>
+    <name>TV Channel</name>
+    <code>0x0504</code>
+    <define>TV_CHANNEL_CLUSTER</define>
+    <client init="false" tick="false">true</client>
+    <server init="false" tick="false">true</server>
+    <description>This cluster provides an interface for controlling the current TV Channel on a device.</description>
+    <attribute side="server" code="0x0000" define="TV_CHANNEL_LIST" type="OCTET_STRING" length="32" writable="false" optional="false">tv channel list</attribute>
+    <attribute side="server" code="0x0001" define="TV_CHANNEL_LINEUP" type="OCTET_STRING" length="32" writable="false" optional="false">tv channel lineup</attribute>
+    <attribute side="server" code="0x0002" define="TV_CHANNEL_CURRENT_CHANNEL" type="OCTET_STRING" length="32" writable="false" optional="false">current tv channel</attribute>
+
+    <command source="client" code="0x00" name="ChangeChannel" optional="false">
+      <description>Change the channel on the TV to the channel case-insensitive exact matching the value passed as an argument. </description>
+      <arg name="match" type="CHAR_STRING"/>
+    </command>
+
+
+    <command source="client" code="0x01" name="ChangeChannelByNumber" optional="false">
+      <description>Change the channel on the TV to the channel with the given Number in the ChannelList attribute.</description>
+      <arg name="majorNumber" type="INT16U"/>
+      <arg name="minorNumber" type="INT16U"/>
+    </command>
+
+    <command source="client" code="0x02" name="SkipChannel" optional="false">
+      <description>This command provides channel up and channel down functionality, but allows channel index jumps of size Count. When the value of the increase or decrease is larger than the number of channels remaining in the given direction, then the behavior SHALL be to return to the beginning (or end) of the channel list and continue. For example, if the current channel is at index 0 and count value of -1 is given, then the current channel should change to the last channel.</description>
+      <arg name="Count" type="INT16U"/>
+    </command>
+
+    <command source="server" code="0x00" name="ChangeChannelResponse" optional="false">
+      <description>Upon receipt, this SHALL display the active status of the input list on screen.</description>
+      <arg name="ChannelMatch" type="TvChannelInfo" array="true"/>
+      <arg name="ErrorType" type="TvChannelErrorType"/>
+    </command>
+
+  </cluster>
+
+  <struct name="TvChannelInfo">
+    <item name="MajorNumber" type="INT16U"/>
+    <item name="MinorNumber" type="INT16U"/>
+    <item name="Name" type="CHAR_STRING"/>
+    <item name="CallSign" type="CHAR_STRING"/>
+    <item name="AffiliateCallSign" type="CHAR_STRING"/>
+  </struct>
+
+  <struct name="TvChannelLineup">
```

Comment:
should we keep naming as in spec (which uses 'LinupInfo' instead of 'Lineup')?

Reply:
Updated.